protocol/messenger.go

```diff
@@ -1804,6 +1804,87 @@ func (m *Messenger) ChangeGroupChatName(ctx context.Context, chatID string, name
 	return m.addMessagesAndChat(chat, buildSystemMessages([]v1protocol.MembershipUpdateEvent{event}, m.systemMessagesTranslations), &response)
 }
 
+func (m *Messenger) EditGroupChat(ctx context.Context, chatID string, name string, color string, image string) (*MessengerResponse, error) {
+	logger := m.logger.With(zap.String("site", "EditGroupChat"))
+	logger.Info("Editing group chat details", zap.String("chatID", chatID), zap.String("name", name), zap.String("color", color))
+
+	chat, ok := m.allChats.Load(chatID)
+	if !ok {
+		return nil, ErrChatNotFound
+	}
+
+	group, err := newProtocolGroupFromChat(chat)
+	if err != nil {
+		return nil, err
+	}
+
+	signAndProcessEvent := func(m *Messenger, event *v1protocol.MembershipUpdateEvent) error {
+		err := event.Sign(m.identity)
+		if err != nil {
+			return err
+		}
+
+		err = group.ProcessEvent(*event)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	var events []v1protocol.MembershipUpdateEvent
+
+	if chat.Name != name {
+		clock, _ := chat.NextClockAndTimestamp(m.getTimesource())
+		event := v1protocol.NewNameChangedEvent(name, clock)
+		event.ChatID = chat.ID
+		err = signAndProcessEvent(m, &event)
+		if err != nil {
+			return nil, err
+		}
+		events = append(events, event)
+	}
+
+	if chat.Color != color {
+		clock, _ := chat.NextClockAndTimestamp(m.getTimesource())
+		event := v1protocol.NewColorChangedEvent(color, clock)
+		event.ChatID = chat.ID
+		err = signAndProcessEvent(m, &event)
+		if err != nil {
+			return nil, err
+		}
+		events = append(events, event)
+	}
+
+	// TODO add image change
```

Comment:
This still a work in progress?

Reply:
Yes, the next phase is to add chat icon to group chat.